Marlin/src/lcd/ultralcd.cpp

```diff
@@ -936,7 +950,7 @@ void MarlinUI::update() {
           #endif // ENCODER_RATE_MULTIPLIER
 
           encoderPosition += (encoderDiff * encoderMultiplier) / (ENCODER_PULSES_PER_STEP);
-          encoderDiff = 0;
+          encoderDiff = 0;  // Move this and clear always?
```

Comment:
Check to see if moving this down to line 964 (after the `if (encoderPastThreshold || lcd_clicked)` condition) to see if it also fixes the issue, while also not causing the encoder to miss steps. The lcd update loop is called either 10 or 20 times per second, so that should be good enough not to miss 99.9% of multiple-pulses.